ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -604,12 +600,11 @@ getCompletions plId ideOpts CC {allModNamesAsNS, anyQualCompls, unqualCompls, qu
         , enteredQual `T.isPrefixOf` label
         ]
 
-      filtImportCompls = filtListWith (mkImportCompl enteredQual) importableModules
+      filtImportCompls = filtListWith (mkImportCompl enteredQual) $ HM.keys moduleExportsMap
```

Comment:
Strictly speaking this change is not correct in a multi-component project, because the keys of `moduleExportsMap` may include modules that are not importable in the current component. 

For example, in the usual lib and test project, it is common for the test component to have dependencies on test packages (hspec, tasty, quickCheck, etc) which are not available in the lib component.

This also applies to the completions that we generate for identifiers - they should be intersected with the importable modules from `envVisibleModules`

Reply:
I understand the problem, but I am a bit confused by the suggested solution. Won't intersecting with `envVisibleModuleNames` just leave us with what we had in the first place? (which did not include local modules)

Reply:
You are right, I did not look into the definition of `envVisibleModuleNames` and it only includes package modules. 

The `GetKnownTargets` key returns the list of local modules for all the components, you could use that. It will still not be 100% correct since a module from the lib component cannot import a module from the test component, but fixing that can be future work.

Reply:
- Either it's empty (what does the project look like?)
- Or it's returning a Nothing (it shouldn't, but check)
- Or it doesn't work with `useWithStaleFast` for some reason - try just `use`

Reply:
- I am working on the HLS project to test my changes
- It is returning `Nothing` when attempting to auto complete within `Completions.hs`
- `use` also returns `Nothing`

Reply:
I see the problem now. GetKnownTargets is a file-independent rule key so you should call useNoFile (or the equivalent StaleFast version)